jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/core/samplers/BaseSampler.java

```diff
@@ -43,4 +46,28 @@ public interface SamplerChild extends DslTestElement {
 
   }
 
+
+  /**
+   * Allows to apply APDEX configuration to report for exact sampler
+   *
+   * @param satisfied - "satisfied" threshold
+   * @param tolerated - "tolerated threshold
+   * @return the sampler itself as BaseSampler
+   *
+   * @since 0.57
+   */
+  public BaseSampler<T> apdex(Duration satisfied, Duration tolerated) {
```

Comment:
I like from current approach the idea that you can do something like 

```java
httpSampler("http://myservice")
	.apdex(Duration.ofMillis(500), Duration.ofSeconds(1))
```

But this is only used by htmlReporter, and I am not sure adding in every sampler a method that is only used by something that is not used in most of the scenarios (even though it comes with built-in jmeter). Sounds like me that is more tied to htmlReporter than samplers, and having such kind of methods in sampelrs makes the api to loose cohesion.

What other alternatives have you considered?

Some ideas that come to my mind with some potential pros and cons:

* Having in htmlReport some way of configuring samplers thresholds:
  * This is nice because the setting is configured where it is being used (high cohesion), but might make configuration more complex (having to pass to html reporter the sampler as parameter) or loosely coupled (passing the name of the sampler). 
  * This is the first idea that comes to my mind, maybe is the most intuitive/expected for users? 
  * If you want to set an apdex on a sampler then just set a name on it and use same name in hmltReporter to configure it.
Eg:
```java
	String samplerName = "mySampler";
	testPlan(
		threadGroup(1, 2,
			httpSampler(samplerName, "http://myservice")
		)
		htmlReporter()
			.transactionApdexThresholds(samplerName, Duration.ofMillis(500), Duration.ofSeconds(1))
```
* Setting apdex threshold with a children element, like some sort of "assertion".
  * This sounds consistent with the behavior of assertions, since in some way apdex is some sort of "soft" assertion and allows you to easily configure some element tied to a sampler. 
  * It also follows jmeter way of "extension" of sampler logic or checking, which settles some guidelines for future additions on sampler without having the sampler api contract to increase in complexity and lose cohesion. 
  * The main problem I see with this is that it requires some "fictional" element, which might bring some problems by itself, and I don't know if users would expect that. 
  * Additionally, there would be a loose connection between the configuration and where it is used.

Have you evaluated other alternatives?

Reply:
Any comments on alternatives? I see you have added some java doc which makes it a little more explicit.

Reply:
Do you see any particular issue with the first alternative described? If not, I can try adapt current implementation to such design.